sdk/tables/data-tables/src/serialization.ts

```diff
@@ -105,42 +108,51 @@ export function serialize(obj: object): object {
   return serialized;
 }
 
-function getTypedObject(value: any, type: string): any {
+function getTypedObject(value: any, type: string, disableDeserialization: boolean): any {
   switch (type) {
     case Edm.Boolean:
+      return disableDeserialization ? { value, type: "Boolean" } : value;
     case Edm.Double:
+      return disableDeserialization ? { value, type: "Double" } : value;
     case Edm.Int32:
+      return disableDeserialization ? { value, type: "Int32" } : value;
     case Edm.String:
-      return value;
+      return disableDeserialization ? { value, type: "String" } : value;
     case Edm.DateTime:
-      return { value, type: "DateTime" };
+      return disableDeserialization ? { value, type: "DateTime" } : new Date(value);
     case Edm.Int64:
-      return { value, type: "Int64" };
+      return disableDeserialization ? { value, type: "Int64" } : BigInt(value);
     case Edm.Guid:
       return { value, type: "Guid" };
     case Edm.Binary:
-      return base64Decode(value);
+      return disableDeserialization ? { value, type: "Binary" } : base64Decode(value);
     default:
       throw new Error(`Unknown EDM type ${type}`);
   }
 }
 
-export function deserialize<T extends object>(obj: object): T {
+export function deserialize<T extends object = Record<string, any>>(
```

Comment:
it feels weird that we call the flag `disableDeserialization` but then still call a `deserialize` function that does some processing on it. Maybe we can find a different name? Like `disableTypeConversion` or something?